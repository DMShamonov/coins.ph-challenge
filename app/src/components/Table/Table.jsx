// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TablePropsType } from './types.js.flow';

export default function Table({
  className,
  children,
}: TablePropsType): React.Node {
  return (
    <div className={classnames('table', className)}>
      {children}
    </div>
  );
}
