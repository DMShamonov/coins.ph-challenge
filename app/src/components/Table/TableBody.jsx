// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableBodyPropsType } from './types.js.flow';

export default function TableBody({
  className,
  children,
}: TableBodyPropsType): React.Node {
  return (
    <div className={classnames('table__body', className)}>
      {children}
    </div>
  );
}
