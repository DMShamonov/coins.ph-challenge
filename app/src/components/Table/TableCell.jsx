// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableCellPropsType } from './types.js.flow';

export default function TableCell({
  width,
  className,
  children,
}: TableCellPropsType): React.Node {
  return (
    <div className={classnames('table__cell', className)} style={{ width }}>
      {children}
    </div>
  );
}
