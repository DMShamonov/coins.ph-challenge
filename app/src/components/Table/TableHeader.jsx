// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableHeaderPropsType } from './types.js.flow';

export default function TableHeader({
  className,
  children,
}: TableHeaderPropsType): React.Node {
  return (
    <div className={classnames('table__header', className)}>
      {children}
    </div>
  );
}
