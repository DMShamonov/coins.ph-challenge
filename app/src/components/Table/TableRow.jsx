// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableRowPropsType } from './types.js.flow';

export default function TableRow({
  className,
  children,
}: TableRowPropsType): React.Node {
  return (
    <div className={classnames('table__row', className)}>
      {children}
    </div>
  );
}
