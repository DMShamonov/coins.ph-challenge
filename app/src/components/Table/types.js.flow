// @flow

import type { Node } from 'react';

export type TablePropsType = {|
  className?: string,
  children: Node,
|};

export type TableHeaderPropsType = {|
  className?: string,
  children: Node,
|};

export type TableBodyPropsType = {|
  className?: string,
  children: Node,
|};

export type TableRowPropsType = {|
  className?: string,
  children: Node,
|};

export type TableCellPropsType = {|
  width?: string,
  className?: string,
  children?: Node,
|};
