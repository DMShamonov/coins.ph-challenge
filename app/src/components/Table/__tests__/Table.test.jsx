import React from 'react';
import { shallow } from 'enzyme';

import Table from '../Table';

describe('`Table` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Table>__CONTENT__</Table>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Table className="__CLASSNAME__">__CONTENT__</Table>)).toMatchSnapshot();
  });
});
