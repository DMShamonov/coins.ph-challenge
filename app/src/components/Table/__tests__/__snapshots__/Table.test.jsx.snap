// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Table\` component should render correctly with \`className\` prop 1`] = `
<div
  className="table __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`Table\` component should render correctly with default props 1`] = `
<div
  className="table"
>
  __CONTENT__
</div>
`;
